cluster/gce/configure-vm.sh

```diff
@@ -140,15 +140,30 @@ function remove-docker-artifacts() {
   echo "== Finished deleting docker0 =="
 }
 
-# Retry a download until we get it.
+# Retry a download until we get it. Takes a hash and a set of URLs.
 #
-# $1 is the URL to download
+# $1 is the sha1 of the URL, if known. $2+ are the URLs to download.
 download-or-bust() {
-  local -r url="$1"
-  local -r file="${url##*/}"
-  rm -f "$file"
-  until curl --ipv4 -Lo "$file" --connect-timeout 20 --retry 6 --retry-delay 10 "${url}"; do
-    echo "Failed to download file (${url}). Retrying."
+  local -r hash="$1"
+  shift 1
+
+  urls=( $* )
+  while true; do
```

Comment:
I can rename it if you want, but the old function actually had the had same behavior. The old behavior is essentially to block inside the `curl` for some period of time doing retries (6, to be precise), then come up for air and `echo` so you'd see something in the logs - otherwise it wasn't clear what was failing. I kept the retry behavior because it's relatively sane - it'll pound on one geo for a minute or two and then fall to another in a round-robin. There are more complicated options we can consider at some point, like trying one download of the first URL first, then trying them all with retries in parallel (sort of like how IPv4 to IPv6 is handled).

But it's actually always been infinite. There's nothing to do if we can't get the release or the Salt tar down.